### ISSUE TITLE
Using a second transformation item in one day causes the item for reversing it to "lag" to the previous type

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica#12357]
ieahleen: "I have been transformed in a Ghost but I have Sand to buy in my reward column instead of the Opaque Potion..."
ieahleen: "update: Now that I am a snowman, I have the Opaque potion in the reward column. Something went out of sync?"
citrusella: "I cycled through flower, starfish, snowman, ghost, and ghost (no downtime between), and I ended up with the following, even after reloads:
    Flower: Petal-free
    Starfish: Petal-free
    Snowman: Sand
    Ghost: Salt
    Ghost: Opaque
I don't know if it's behind or if something's off or what..."

I could reproduce this. Furthermore, I only tested this once but when I tried to use the Sand reward while I was a Flower, it didn't work. It did let me see the Petal-Free potion which I could then successfully use.

**Comment by PitiTheGrey** [comment]
Trying to reproduce the above mentioned errors, it took me 4 transformation items until I finally got 'the wrong detranformation remedy. I was using chrome browser. What do you think?

**Comment by shanaqui** [comment]
@PitiTheGrey Did you _use_ the reward to transform yourself back? Because I reproduced it by _not_ using the reward and going straight for each transition (just observing which reward is available), and it's reliable for me: it happens every time.

**Comment by DJohnson23** [comment]
Hi - can I pick up this defect?

**Comment by PitiTheGrey** [comment]
Yes. I am not working on it any more. CU PitiVon meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: DJohnson23 ***@***.***> Datum: 08.04.23  18:59  (GMT+01:00) An: HabitRPG/habitica ***@***.***> Cc: PitiTheGrey ***@***.***>, Mention ***@***.***> Betreff: Re: [HabitRPG/habitica] Using a second transformation item in one day causes the item for reversing it to "lag" to the previous type (#12357) 
Hi - can I pick up this defect?

**Comment by CuriousMagpie** [comment]
Sure thing, @DJohnson23! Thanks!
